### ISSUE TITLE
iViewer UI

### ISSUE THREAD

**Issue by will-moore** [openmicroscopy/design#75]
Some proposed cleanup & future feature prototyping for iViewer.
Generally aiming to reduce 'ink' on the page to give a cleaner look.

cc @waxenegger @jburel 

![screen shot 2017-04-25 at 16 25 23](https://cloud.githubusercontent.com/assets/900055/25394378/c7d96908-29d5-11e7-8410-5dbf9d625cf2.png)

![screen shot 2017-04-25 at 16 31 02](https://cloud.githubusercontent.com/assets/900055/25394392/cdee31e8-29d5-11e7-9b4f-701f362dfc6d.png)

**Comment by jburel** [comment]
scale bar location has been raised as a problem so we should try to keep it in the "bottom-left corner" of the image not the viewport

**Comment by jburel** [comment]
To avoid confusion, User settings with the list of thumbnails will still be displayed in "settings"

**Comment by jburel** [comment]
Discussed today: look into making the scalebar draggable

**Comment by waxenegger** [comment]
Should the dataset image side panel on the left still be resizable/collapsible?

**Comment by waxenegger** [comment]
@jburel  and @will-moore 

Is the only way to turn off rois the checkbox on the header of the table....

This is going to be a bit of a technical nightmare, the reason being that instead of being able to turn off the vector layer altogether in ol3, I keep it on but make all shapes "invisible" (by tricking ol3 with filtering). Essentially all the workflow and code paths will be still active. Originally the visibility was more a means to turn off individual shapes but as a means to not hide the rois altogether it's not a great solution but will be one that's inevitable. There are various events in place to guarantee that after some action such as selection or any modification the state of the shapes are sync with the embedded ol3 viewer. Now what that means is that I cannot allow for certain actions such as delete or modify or copy and paste or attachment locks without having the vector layer on. shapes or features as their counterpart in ol3 are inextricably linked to the vector layer and that makes good sense because without that layer there is no means of working with them.

So if we plan on showing the table while every shape in it is not visible to the user allowing for some actions on invisible rois, the only one that really makes sense to me being delete, not sure of copy & pasting and modifying invisible shapes....then I'm going to have to disable some actions at a minimum.

Another thing is that the columns of the roi table are actually resizable and the checkbox can be pushed over.

**Comment by will-moore** [comment]
As discussed, unchecking the Show in table header will simply hide every shape in the viewer but you can still interact with them in the table, delete them etc. If a user draws a new shape it will be visible and the header Show will become checked (same as if user manually checks Show on one or more shapes).

**Comment by jburel** [comment]
That is not what is currently implemented.
The Show in the header is only ticked if all checkboxes are checked

**Comment by will-moore** [comment]
Some ideas on how to show more measurements and/or annotations on Shapes:

Small changes: move Show column to left, re-order toolbars so that those showing/editing selected shapes are closer to the shapes themselves. Also adds a 'wrench' to popup a checklist of which columns to show.
![screen shot 2017-06-23 at 14 21 07](https://user-images.githubusercontent.com/900055/27484066-b7135d54-581f-11e7-95f3-2ea35d432cc8.png)

Choose to show Length and Area for shapes.
NB: we don't show Comments, so the Comments toolbar is not shown.

![screen shot 2017-06-23 at 14 21 07](https://user-images.githubusercontent.com/900055/27485544-0adfec04-5825-11e7-9470-3102af44f1ba.png)

Now we show Tags column. This shows a summary of tags for each shape and also adds the Tags toolbar where we can see full tag list and Add / Remove tags. Using bootstrap's "badges" to style tags. Can show more info on a tool-tip for each tag (at least in the toolbar - maybe in the table too?). TODO: add a "remove" icon for each tag in the toolbar.

![screen shot 2017-06-23 at 14 21 27](https://user-images.githubusercontent.com/900055/27485571-2d3c259c-5825-11e7-908a-610a9d0875e8.png)

cc @jburel @waxenegger

**Comment by jburel** [comment]
For any measurements we will need units and maybe ability to switch between for example pixels and microns

**Comment by pwalczysko** [comment]
Export intensities in the ROI to somewhere - is that planned to be in a new tab ?

**Comment by jburel** [comment]
Idea is to export measurement as csv
I don't think displaying the intensity as it is in insight is that useful

**Comment by pwalczysko** [comment]
@jburel : Agreed. I was just thinking about where  the export button would live.

**Comment by jburel** [comment]
We will also need a way to trigger the analysis of the selected shapes
In that case, do we clean the results already displayed for the shapes not in the "active" selection?

**Comment by will-moore** [comment]
Probably add an "Export/Download" button beside the Save/Undo/Redo buttons at the top.
But I'm not sure how we'd display pixel intensities differently from shape dimensions. I think we would want to display them in the shapes table in the same way (not *just* download them) and this wouldn't be too hard if the server could give us intensities for specified shapes. It gets tricky when we have a large number of shapes or they cover a large area. Maybe we can use cut-offs to say "If more than 50 shapes or more than 1 000 000 pixels in total area then we're not going to show you intensities and you have to download them"?

**Comment by pwalczysko** [comment]
@will-moore : Intensities ? You mean to show average intensity per ROI ? You would not have a space for showing even all intensities from relatively small ROIs ?

**Comment by jburel** [comment]
Export might be more suitable since we were thinking of having a Download button on the info tab
